src/ucp/core/ucp_ep.c

```diff
@@ -1477,23 +1486,28 @@ ucs_status_t ucp_ep_config_init(ucp_worker_h worker, ucp_ep_config_t *config,
 
         if (rsc_index != UCP_NULL_RESOURCE) {
             iface_attr = ucp_worker_iface_get_attr(worker, rsc_index);
+            md_attr    = &context->tl_mds[context->tl_rscs[rsc_index].md_index].attr;
 
             /* GET Zcopy */
             if (iface_attr->cap.flags & UCT_IFACE_FLAG_GET_ZCOPY) {
-                scale = ucp_tl_iface_bandwidth(context, &iface_attr->bandwidth) /
-                        rndv_max_bw;
-                if (scale < (1. / context->config.ext.multi_lane_max_ratio)) {
-                    continue;
+                ucs_for_each_bit(mem_type_index, md_attr->cap.reg_mem_types) {
```

Comment:
I don't think this is correct - in this for_each loop we set `config->tag.rndv.scale[lane]` and `config->tag.rndv.get_zcopy_lanes[get_zcopy_lane_count++]` values several times = # of memory type MDs.
so, it makes `get_zcopy_lane_count` greater than we really have

should we have `min_get_zcopy`/`max_get_zcopy`/`get_zcopy_lanes`/`scale` for each memory type? then it should work

Reply:
i have `"break"` at the line :1509 . the lane will be added only one. am I missing something?

Reply:
@bureddy thanks, I see now. didn't notice this `break` operator
does it mean if we have two lanes configured for intra-node RNDV(`knem`, `cuda_ipc`), before this fix `knem` wasn't selected due to low performance in comparison to `cuda_ipc`? so, we didn't perform RNDV for HOST memory buffers.

Reply:
correct